pkg/executor/infoschema_reader.go

```diff
@@ -546,25 +546,18 @@ func fetchColumnsFromStatsCache(table *model.TableInfo) (rowCount uint64, avgRow
 	return
 }
 
-func (e *memtableRetriever) updateStatsCacheIfNeed(sctx sessionctx.Context) (bool, error) {
+func (e *memtableRetriever) updateStatsCacheIfNeed() bool {
 	for _, col := range e.columns {
 		// only the following columns need stats cahce.
 		if col.Name.O == "AVG_ROW_LENGTH" || col.Name.O == "DATA_LENGTH" || col.Name.O == "INDEX_LENGTH" || col.Name.O == "TABLE_ROWS" {
-			err := cache.TableRowStatsCache.Update(sctx)
-			if err != nil {
-				return false, err
-			}
-			return true, err
+			return true
```

Comment:
Because stat cache update costs a lot of time, stat cache update every 3 seconds.
Only need to update stat cache with returned table id.

Reply:
8.51s-->0.06s
With opt:
Before: 
340 rows in set (8.51 sec)
After: 
1. 340 rows in set (3.32 sec): updated all stat row cache
2. 340 rows in set (0.06 sec): not update stat row cache.
The time varies.
Because stat cache update costs a lot of time, stat cache update every 3 seconds
Need to update stat cache with returned table id.
https://github.com/pingcap/tidb/pull/50874/commits/f3bf37bd6c57fc3726a8c377965311f25ece4728
With the above change, time stabled.